tasks/pin_node_group.rb

```diff
@@ -17,12 +21,13 @@
 http.key = OpenSSL::PKey::RSA.new(File.read(Puppet.settings['hostprivkey']))
 http.ca_file = Puppet.settings['localcacert']
 
-# Get the id of the 'PE Master' node group
+# Get the id of the node group
 res = http.request_get(uri.request_uri).read_body
-master_id = JSON.parse(res).find { |group| group['name'] == 'PE Master' }['id']
+node_group_id = JSON.parse(res).find { |group| group['name'] == node_group }['id']
+raise "No group ID could be found for #{node_group}" if node_group_id.nil?
 
 # use the /pin endpoint to pin our compile masters to PE Master
```

Comment:
Can you also generalize this comment?

Reply:
@m0dular I finally got back to this. Let me know if that works for you.